pkg/dockerregistry/server/auth.go

```diff
@@ -284,6 +284,9 @@ func (ac *AccessController) Authorized(ctx context.Context, accessRecords ...reg
 				if !isMetricsBearerToken(ac.metricsConfig, bearerToken) {
 					return nil, ac.wrapErr(ctx, ErrOpenShiftAccessDenied)
 				}
+				if err := verifyMetricsAccess(ctx, osClient); err != nil {
+					return nil, ac.wrapErr(ctx, err)
+				}
```

Comment:
You need to combine this w/ the isMetricsBearerToken check...right now you're going to return an access denied before you even get here, right?

Reply:
@bparees If the secret token is not true, then it makes no sense to continue. At the moment, a secret token is always required.

Reply:
Hm, i would have expected us to be deprecating the secret token and not requiring it.

(Such that in 3.10 you can either supply the (deprecated) secret token, or a valid user auth header)

Reply:
@bparees But I understand what you mean. It is necessary to support two options: with a secret and with a role. right ?

Reply:
see above.  I'd expect us to deprecate the "with a secret" path.  What does the router offer?

Reply:
@bparees You can set a static username/password or SAR check.

https://github.com/openshift/origin/blob/master/pkg/router/metrics/metrics.go#L52

Reply:
Ok, I still think we should support "either/or" not "and" then.